daemonize/src/lib.rs

```diff
@@ -166,6 +168,14 @@ impl From<File> for Stdio {
     }
 }
 
+impl From<RawFd> for Stdio {
+    fn from(fd: RawFd) -> Self {
+        Self {
+            inner: StdioImpl::RedirectToRawFd(fd),
+        }
+    }
+}
```

Comment:
You may want to look at preferring the safe equivalents that are available now? https://github.com/sunfishcode/io-lifetimes/issues/38

> _If a type implements `AsRawFd`, `FromRawFd`, and/or `IntoRawFd`, it usually wants `AsFd`, `From<OwnedFd>`, and/or `From<T> for OwnedFd`, respectively._

Since Rust 1.63.0 (Aug 2022), [`OwnedFd`](https://doc.rust-lang.org/stable/std/os/fd/struct.OwnedFd.html) became available, which provides `as_fd()` and `as_raw_fd()` if you need to get `BorrowedFd` or `RawFd` types (_[original RFC](https://github.com/sunfishcode/rfcs/blob/main/text/0000-io-safety.md#ownedfd-and-borrowedfdfd) for further context_).

```suggestion
impl From<OwnedFd> for Stdio {
    fn from(fd: OwnedFd) -> Self {
        Self {
            inner: StdioImpl::RedirectToFd(fd),
        }
    }
}
```

- You would also need to change the import from `use std::os::unix::io::RawFd;` to `use std::os::fd::OwnedFd;`
- There's also a `AsRawFd` import and a `as_raw_fd()` call for the `RedirectToFile` enum variant, that can [get an `OwnedFd` since `1.63.0` in a similar way](https://doc.rust-lang.org/stable/std/convert/trait.From.html#impl-From%3CFile%3E-for-OwnedFd).

---

That said, I'm not quite sure why `Stdio` struct exists here instead of using the [`std::process::Stdio`](https://doc.rust-lang.org/stable/std/process/struct.Stdio.html) one? It already supports `From` for file and fd, while there is a separate function call `null()` for getting the equivalent `/dev/null` and the `Keep` enum variant seems to be for a No-op.

Ah.. so this is for redirection of the streams handled in the `redirect_standard_streams()` call further down. So at the time Rust didn't have an equivalent built-in for `Stdio` but it does now since Rust 1.74 (Nov 2023). So if raising the MSRV were viable, one could switch to that to simplify 😎